gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -266,23 +264,28 @@ private static void put(ByteString source, int offset, int size, ByteBuffer dest
   private int readBufferedContentInto(ByteBuffer byteBuffer) {
     // Handle skipping forward through the buffer for a seek.
     long bytesToSkip = positionForNextRead - positionInGrpcStream;
-    long bufferSkip = min(bufferedContent.size() - bufferedContentReadOffset, bytesToSkip);
-    bufferSkip = max(0, bufferSkip);
-    bufferedContentReadOffset += bufferSkip;
-    positionInGrpcStream += bufferSkip;
-    int remainingBufferedBytes = bufferedContent.size() - bufferedContentReadOffset;
-
-    boolean remainingBufferedContentLargerThanByteBuffer =
-        remainingBufferedBytes > byteBuffer.remaining();
+
+    if (bytesToSkip >= bufferedContent.size()) {
+      positionInGrpcStream += bufferedContent.size();
+      invalidateBufferedContent();
+      return 0;
+    }
+
+    if (bytesToSkip > 0) {
+      positionInGrpcStream += bytesToSkip;
+      bufferedContent = bufferedContent.substring(Math.toIntExact(bytesToSkip));
+    }
+
+    boolean bufferedContentLargerThanByteBuffer = bufferedContent.size() > byteBuffer.remaining();
```

Comment:
nit: What about using below implementation? We can skip maintaining `bufferedContentLargerThanByteBuffer`.
```
int bytesToWrite = Math.min( bufferedContent.size() , byteBuffer.remaining());

put(bufferedContent, 0, bytesToWrite, byteBuffer);
positionInGrpcStream += bytesToWrite;
positionForNextRead += bytesToWrite;

if(bytesToWrite == bufferedContent.size()) {
invalidateBufferedContent();
} else {
bufferedContent = bufferedContent.substring(bytesToWrite);
}
```